src/App.test.jsx

```diff
@@ -1,14 +1,62 @@
-import { render } from '@testing-library/react';
+import { fireEvent, render } from '@testing-library/react';
+
+import { useSelector, useDispatch } from 'react-redux';
 
 import App from './App';
 
-test('App', () => {
-  const { getByText } = render((
-    <App />
-  ));
+jest.mock('react-redux');
+
+describe('App', () => {
+  const tasks = [
+    {
+      id: 1,
+      title: '아무 것도 하지 않기',
+    },
+    {
+      id: 2,
+      title: '그래도 뭐라도 하기',
+    },
+  ];
+
+  useSelector.mockImplementation((selector) => selector({
+    taskTitle: 'new title',
+    tasks,
+  }));
+
+  const dispatch = jest.fn();
+  useDispatch.mockImplementation(() => dispatch);
+
+  function customRender() {
+    return render((
+      <App />
+    ));
+  }
+
+  it('renders to do list', () => {
+    const { queryByText } = customRender();
+
+    expect(queryByText(/아무 것도 하지 않기/)).not.toBeNull();
+    expect(queryByText(/그래도 뭐라도 하기/)).not.toBeNull();
+  });
+
+  it('removes completed todo item', () => {
+    const { getAllByText } = customRender();
+    fireEvent.click(getAllByText(/완료/)[0]);
+
+    expect(dispatch).toBeCalledWith({ type: 'deleteTask', payload: { id: 1 } });
+  });
+
+  it('adds todo item', () => {
+    const { getByText } = customRender();
+    fireEvent.click(getByText(/추가/));
+
+    expect(dispatch).toBeCalledWith({ type: 'addTask' });
+  });
 
-  expect(getByText(/추가/)).not.toBeNull();
+  it('changes title', () => {
+    const { getByLabelText } = customRender();
+    fireEvent.change(getByLabelText(/할 일/), { target: { value: 'a' } });
```

Comment:
빈칸을 넣어주시면 가독성이 더 좋을거 같아요 :)
```suggestion
    const { getByLabelText } = customRender();
    
    fireEvent.change(getByLabelText(/할 일/), { target: { value: 'a' } });
```

Reply:
넵